packages/core-v2-sdk/src/models/Term.ts

```diff
@@ -47,4 +48,27 @@ export class Term {
   getYieldToken(startTimeStamp: number): YieldToken {
     return new YieldToken(this.id, this.context, this);
   }
+
+  /**
+   * Gets the time remaining of the term in seconds. If expired, returns zero.
+   * @async
+   * @return {Promise<number>} time remaining in seconds
+   */
+  async getSecondsUntilExpiry(): Promise<number> {
```

Comment:
Same as above, but also since this is repeated logic, it might be worth pulling out into a util function that takes `expiryInSeconds: number, atBlock?: number` arguments.